flash/image/classification/adapters.py

```diff
@@ -0,0 +1,391 @@
+        self,
+        dataset: BaseAutoDataset,
+        batch_size: int = 1,
+        num_workers: int = 0,
+        pin_memory: bool = False,
+        collate_fn: Callable = lambda x: x,
+        shuffle: bool = False,
+        drop_last: bool = True,
+        sampler: Optional[Sampler] = None,
+    ) -> DataLoader:
+        assert batch_size == 1
+        return super().process_predict_dataset(
+            self.convert_dataset(dataset, collate_fn),
+            batch_size,
+            num_workers,
+            pin_memory,
+            collate_fn,
+            shuffle,
+            drop_last,
+            sampler,
+        )
+
+
+class DefaultAdapter(Adapter):
+    """The ``DefaultAdapter`` is an :class:`~flash.core.adapter.Adapter`."""
+
+    required_extras: str = "image"
+
+    def __init__(self, task: AdapterTask, backbone: torch.nn.Module, head: torch.nn.Module):
+        super().__init__()
+
+        self._task = NoModule(task)
+        self.backbone = backbone
+        self.head = head
+
+    @property
+    def task(self) -> Task:
+        return self._task.task
+
+    @classmethod
+    @catch_url_error
+    def from_task(
+        cls,
+        *args,
+        task: AdapterTask,
+        backbone: torch.nn.Module,
+        head: torch.nn.Module,
+        **kwargs,
+    ) -> Adapter:
+        return cls(task, backbone, head)
+
+    def training_step(self, batch: Any, batch_idx: int) -> Any:
+        batch = (batch[DefaultDataKeys.INPUT], batch[DefaultDataKeys.TARGET])
+        return Task.training_step(self.task, batch, batch_idx)
+
+    def validation_step(self, batch: Any, batch_idx: int) -> Any:
+        batch = (batch[DefaultDataKeys.INPUT], batch[DefaultDataKeys.TARGET])
+        return Task.validation_step(self.task, batch, batch_idx)
+
+    def test_step(self, batch: Any, batch_idx: int) -> Any:
+        batch = (batch[DefaultDataKeys.INPUT], batch[DefaultDataKeys.TARGET])
+        return Task.test_step(self.task, batch, batch_idx)
+
+    def predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int = 0) -> Any:
+        batch[DefaultDataKeys.PREDS] = Task.predict_step(
+            self.task, (batch[DefaultDataKeys.INPUT]), batch_idx, dataloader_idx=dataloader_idx
+        )
+        return batch
+
+    def forward(self, x) -> torch.Tensor:
+        x = self.backbone(x)
+        if x.dim() == 4:
+            x = x.mean(-1).mean(-1)
+        return self.head(x)
+
+
+def _fn():
+    pass
+
+
+TRAINING_STRATEGIES = FlashRegistry("training_strategies")
+TRAINING_STRATEGIES(name="default", fn=_fn, adapter=DefaultAdapter, algorithm=str)
+
+if _LEARN2LEARN_AVAILABLE:
+    from learn2learn import algorithms
+
+    for algorithm in dir(algorithms):
+        try:
+            if "lightning" in algorithm.lower() and issubclass(getattr(algorithms, algorithm), LightningModule):
+                TRAINING_STRATEGIES(
+                    name=algorithm.lower().replace("lightning", ""),
+                    fn=_fn,
```

Comment:
We could perhaps just do `fn=partial(Learn2LearnAdapter, algorithm=getattr(algorithms, algorithm))` here, then not need to get the metadata later on